docs/decisions/0009-moving-catalog-diffing-to-enterprise-catalog.rst

```diff
@@ -0,0 +1,89 @@
+an onboarding perspective but also from a debugging point of view. Log noise and number of potential error sources means
+tracking down issues within the Integrated Channels are costly and time consuming.
+
+How do the Integrated Channels currently handle customer catalog data?
+======================================================================
+
+existing flow chart:
+
+.. image:: ../images/0009-moving-catalog-diffing-to-enterprise-catalog.rst
+  :width: 500
+
+Currently on a scheduled job, the Integrated Channels' content metadata exporter iterates over each customer's catalog
+and builds up a dict of content representing the current state of the customer's catalog content. Take away- as it
+stands now, the Integrated Channels assumes that the content metadata exporter will report the entirety of the
+customer's purchased content, then pass that payload to the content metadata transmitter which compares the payload with
+all saved `ContentMetadataItemTransmission` items under the customer to determine which items it (the transmitter) needs
+to create, update, and delete. It then chunks up each of the buckets, and sends those chunks (up to a limited number of
+times) to the client to send to the customer's LMS and saves the updated `ContentMetadataItemTransmission` entry.
+
+How the Integrated Channels could better determine if customer metadata needs updating
+======================================================================================
+
+proposed flow chart:
+
+.. image:: ../images/new-integrated-channels-metadata-flow.png
+  :width: 500
+
+If we build an endpoint that takes catalog UUIDs and a set of content keys linked to the uuid and returns three buckets
+of data:
+
+1) What content keys exist under the specified catalog that were not provided in the list of content keys
+
+2) What content keys don't exist under the specified catalog that were provided in the list of content keys
+
+3) What the last updated at times of the content keys are for the content keys provided that are under the specified
+catalog
+
+We would be able to move the responsibility of diffing what the customer has with what the customer should have to the
+enterprise catalog. This would mean that the only job of the Integrated Channels would be to compare the last updated
+times of the courses that already exist to get which courses need updates. Creates and deletes would already determined
+by the first endpoint.
+
+Further Improvements
+====================
+
+There are two additional improvements to explore with the Integrated Channels' metadata transmission flow. Firstly,
+we remove records from our transmission audit table when a delete request is issued. This can result in `lost` nodes of
+content (as we've actively seen from customers) where content can be assumed to be deleted on our end but have it exist
+still on the customer's external LMS. The fix here is pretty easy- if we choose to not delete records, but rather create
```

Comment:
We really should consider doing this asap -- imho this is "need to have" versus "nice to have".